src/geo/transform.js

```diff
@@ -1306,6 +1310,48 @@ class Transform {
         return this._mercatorZfromZoom(this._minZoomForBounds());
     }
 
+    _calcFogMatrices() {
+        if (!this.fogEnd || !this.fogCulling) {
+            return;
+        }
+
+        const maxVisibleAltitude = this.elevation ? this.elevation.visibleDemTiles.reduce((res, tile) => {
+            if (tile.dem) {
+                const tree = tile.dem.tree;
+                const maxMip = tree.maximums[0];
+                res = Math.max(res, maxMip);
+            }
+            return res;
+        }, 0) * this.elevation.exaggeration() : 0;
+
+        const projMaxVisibleAltitude = (maxVisibleAltitude * this.pixelsPerMeter) / Math.cos(this._pitch);
```

Comment:
If merged, it would be nice if this could be traced back to a PR with even just a picture (if not a fully, detailed description of the math). I think this reaches a critical density of trig functions and projections that it will be quite difficult to reverse engineer the math.

Reply:
Related: onto what is this a projection? It seems like maybe a projection of an altitude onto the horizontal plane as the camera pitches?

Reply:
Yes agreed, I'm working on a simplified version of that, which should be easier to follow.